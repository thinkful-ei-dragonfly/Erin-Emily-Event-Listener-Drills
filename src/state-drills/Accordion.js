import React from 'react';

export default class Accordion extends React.Component {
  constructor () {
    super()
    this.handleShowContent = this.handleShowContent.bind(this);
  }
  
  static defaultProps = {
    sections: []
  }

  state = {
    activeButton: null,
  }

  handleShowContent = (sectionIndex) => {
    this.setState({
      activeButton: sectionIndex
    })
    console.log(this.state.activeButton);
  }

  render() {
    const {activeButton} = this.state
    const mappedSections = this.props.sections.map(function(item, index) { 
    // item.id = index;
    if (index === activeButton) {
    return <li key={index}><button onClick = {() => this.handleShowContent(index)}>{item.title}</button><p>hello</p></li>
  } else {
    return <li key={index}><button onClick = {() => this.handleShowContent(index)}>{item.title}</button></li>
  }})
    console.log(mappedSections);
    return (
      <ul>
        {mappedSections}
      </ul>
    )
  }

}