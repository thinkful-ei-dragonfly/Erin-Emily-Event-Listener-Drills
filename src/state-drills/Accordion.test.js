import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
import Accordion from './Accordion';
import { exportAllDeclaration } from '@babel/types';

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe(`Accordion Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty li element if no props', () => {
    const tree = renderer
      .create(<Accordion />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders no sections as active by default', () => {
    const tree = renderer
      .create(<Accordion sections={sectionsProp}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders section content when component is clicked', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp}/>)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders only the last section open when multiple are clicked', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp}/>)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })

});