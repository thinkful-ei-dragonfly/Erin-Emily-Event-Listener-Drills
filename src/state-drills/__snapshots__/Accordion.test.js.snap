// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty li element if no props 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;
